### PR TITLE
fix(card): additional card size names

### LINKED ISSUE #3283: [Card] Additional sizes and names
<ins style="box-sizing: border-box; margin-top: 0px !important; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><h3 class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; margin-left: 16px;">What package is this for?</h3></ins><ul class="contains-task-list added rich-diff-level-zero" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; background: var(--color-success-subtle); text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><li class="task-list-item rich-diff-level-one" style="box-sizing: border-box; list-style-type: none; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;"><input type="checkbox" id="" disabled="" class="task-list-item-checkbox" checked="" style="box-sizing: border-box; font: inherit; margin: 0px 0.2em 0.25em -1.6em; overflow: visible; padding: 0px; vertical-align: middle; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;"><label style="box-sizing: border-box; font-weight: 400;"></label><span> </span>React</li><li class="task-list-item rich-diff-level-one" style="box-sizing: border-box; list-style-type: none; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;"><input type="checkbox" id="" disabled="" class="task-list-item-checkbox" style="box-sizing: border-box; font: inherit; margin: 0px 0.2em 0.25em -1.6em; overflow: visible; padding: 0px; vertical-align: middle; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;"><label style="box-sizing: border-box; font-weight: 400;"></label><span> </span>Angular</li></ul><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><h3 class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; margin-left: 16px;">Summary</h3></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;"><a href="https://pages.github.ibm.com/ai-applications/design/components/card/usage/" rel="nofollow" class="rich-diff-level-one" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;">PAL card guidance</a></p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">In the PAL, the card component has 5 card sizes and a different naming structure than what is in React. This was due to the card existing in code before with no documentation on the site, and then the dashboard guild working on cards later on.</p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">I've updated the guidance to align on the sizes available in React, but this issue is to also include 2 additional card sizes, and to update the naming convention we have for the card widths and heights.</p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">After talking to<span> </span><a class="user-mention notranslate rich-diff-level-one" data-hovercard-type="user" data-hovercard-url="/users/davidicus/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/davidicus" style="box-sizing: border-box; background-color: transparent; color: var(--color-fg-default); text-decoration: none; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color; font-weight: 600; white-space: nowrap;">@davidicus</a><span> </span>about it, we won't be replacing the existing names for those using it, but adding additional names to the card sizes.</p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;"><a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/43549567/153619470-c76e0346-38d6-4f2d-a333-a996c34e5517.png" class="rich-diff-level-one" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;">card-sizes-2</a></p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">Card widths:</p></ins><ul class="added rich-diff-level-zero" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; background: var(--color-success-subtle); text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Extra small: 4 columns</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Small: 6 columns (new size)</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Medium: 8 columns</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Large: 12 columns (new size)</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Extra large: 16 columns</li></ul><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">Card heights:</p></ins><ul class="added rich-diff-level-zero" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; background: var(--color-success-subtle); text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px;"><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Short: 144 px</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Regular: 304 px</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Tall: 464 px (new size)</li><li class="rich-diff-level-one" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px; margin-left: 16px;">Extra tall: 624 px</li></ul><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">Based on what I see in the [storybook], the 1:1 mapping for these new names would be:</p></ins><ins style="box-sizing: border-box; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);">

Existing name | New name
-- | --
SMALL | Extra small, Short
SMALLWIDE | Medium, Short
SMALLFULL | Extra large, Short
MEDIUMTHIN | Extra small, Regular
MEDIUM | Medium, Regular
MEDIUMWIDE | Extra large, Regular
LARGETHIN | Extra small, Extra tall
LARGE | Medium, Extra tall
LARGEWIDE | Extra large, Extra tall

</ins><ins style="box-sizing: border-box; margin-bottom: 0px !important; box-shadow: inset 4px 0 0 var(--color-success-muted); display: block; border-radius: 0px; border-top: 0px; border-bottom: 0px; text-decoration: none; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23);"><p class="rich-diff-level-zero" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; margin-left: 16px;">Would it make sense to separate out the width and height sizes for more flexibility?</p></ins>What package is this for?
 React
 Angular
Summary
[PAL card guidance](https://pages.github.ibm.com/ai-applications/design/components/card/usage/)

In the PAL, the card component has 5 card sizes and a different naming structure than what is in React. This was due to the card existing in code before with no documentation on the site, and then the dashboard guild working on cards later on.

I've updated the guidance to align on the sizes available in React, but this issue is to also include 2 additional card sizes, and to update the naming convention we have for the card widths and heights.

After talking to @davidicus about it, we won't be replacing the existing names for those using it, but adding additional names to the card sizes.

card-sizes-2

Card widths:

Extra small: 4 columns
Small: 6 columns (new size)
Medium: 8 columns
Large: 12 columns (new size)
Extra large: 16 columns
Card heights:

Short: 144 px
Regular: 304 px
Tall: 464 px (new size)
Extra tall: 624 px
Based on what I see in the [storybook], the 1:1 mapping for these new names would be:

Existing name	New name
SMALL	Extra small, Short
SMALLWIDE	Medium, Short
SMALLFULL	Extra large, Short
MEDIUMTHIN	Extra small, Regular
MEDIUM	Medium, Regular
MEDIUMWIDE	Extra large, Regular
LARGETHIN	Extra small, Extra tall
LARGE	Medium, Extra tall
LARGEWIDE	Extra large, Extra tall
Would it make sense to separate out the width and height sizes for more flexibility?

### PR DESCRIPTION
**Pull request by jessieyan**
Closes #3283 

**Summary**

- added SMALLTHICK, MEDIUMTHICK and LARGETHICK sizes

image

**Change List (commits, features, bugs, etc)**

- added new card sizes in LayoutConstants.jsx

**Acceptance Test (how to verify the PR)**

- tests here

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/DashboardEditor/DashboardEditor.test.e2e.jsx [hunk 1]
```diff
@@ -184,7 +184,7 @@ describe('DashboardEditor', () => {
         expect(mockOnCardChange).to.be.calledWith(
           {
             ...card,
-            size: CARD_SIZES.MEDIUMWIDE,
+            size: CARD_SIZES.MEDIUMTHICK,
```
Comment (davidicus) [inline packages/react/src/components/DashboardEditor/DashboardEditor.test.e2e.jsx:187]: What is the reason we are changing this test?

File: packages/react/src/components/Dashboard/DashboardGrid.test.e2e.jsx [hunk 2]
```diff
@@ -174,7 +174,7 @@ const DashboardAllCardsAsResizable = ({ breakpoint, type }) => {
     const hasLinks = index === 4;
     const hasSort = index === 1;
     const hasThresholds = index === 2;
-    const hasExpandedRows = index === 3;
+    const hasExpandedRows = index === 5;
```
Comment (davidicus) [inline packages/react/src/components/Dashboard/DashboardGrid.test.e2e.jsx:177]: Not following why this change is along with the card size changes?

**Review by davidicus — COMMENTED** [review]
I am not sure what caused your visual snapshots to change so drastically? Any ideas. Especially the Table snapshot seems odd.

File: packages/react/src/components/DashboardEditor/DashboardEditor.test.e2e.jsx (hunk 1 above)
Comment (jessieyan) [reply packages/react/src/components/DashboardEditor/DashboardEditor.test.e2e.jsx:187]: The is a resize test. Previously, after resize `MEDIUM` should resize to `MEDIUMWIDE` which is the next one. But now we introduce `MEDIUMTHICK` between `MEDIUM` and `MEDIUMWIDE`, resize goes to `MEDIUMTHICK` as the next one. 

Cypress failed with errors like this:
image

File: packages/react/src/components/Dashboard/DashboardGrid.test.e2e.jsx (hunk 2 above)
Comment (jessieyan) [reply packages/react/src/components/Dashboard/DashboardGrid.test.e2e.jsx:177]: This is due to the added sizes "SMALLTHICK", "MEDIUMTHICK", "LARGETHICK". The index is base on the index in CARD_SIZES. `const hasExpandedRows = index === 3;` was supposed to work with `LARGETHIN` in the test. The new index for `LARGETHIN` is 5.  

I updated the test to omit "SMALLTHICK", "MEDIUMTHICK" for now so that we don't have to change the index here anymore.

**Review by davidicus — APPROVED** [review]
